docs/docs/mapping/customizing_your_gateway.md

```diff
@@ -358,3 +358,34 @@ HTTP statuses and their mappings to gRPC statuses:
 - HTTP `400 Bad Request` -> gRPC `3 INVALID_ARGUMENT`
 
 This method is not used outside of the initial routing.
+
+### Customizing Routing Errors
+
+If you want to retain HTTP `405 Method Not Allowed` instead of allowing it to be converted to the equivalent of the gRPC `12 UNIMPLEMENTED`, which is  HTTP `501 Not Implmented` you can use the following example:
+
+```go
+func handleRoutingError(ctx context.Context, mux *ServeMux, marshaler Marshaler, w http.ResponseWriter, r *http.Request, httpStatus int) {
+	var stderr error
+
+	switch httpStatus {
+	case http.StatusMethodNotAllowed:
+		sterr = HTTPStatusError{
+			HTTPStatus: httpStatus
+			Err:        status.Error(codes.Unimplemented, http.StatusText(httpStatus))
+		}
+	default:
+		runtime.DefaultRoutingErrorHandler(ctx, mux, marshaler, writer, request, httpStatus)
+
+		return
+	}
+
+	runtime.DefaultHTTPErrorHandler(ctx, mux, marshaler, w , r, stderr)
```

Comment:
This might just be me but this seems unnecessarily elaborate. How about this:
```suggestion
	if httpStatus != http.StatusMethodNotAllowed {
		runtime.DefaultRoutingErrorHandler(ctx, mux, marshaler, writer, request, httpStatus)
	}
	
	// Use HTTPStatusError to customize the DefaultHTTPErrorHandler status code
	err := &HTTPStatusError{
		HTTPStatus: httpStatus
		Err:        status.Error(codes.Unimplemented, http.StatusText(httpStatus))
	}
	runtime.DefaultHTTPErrorHandler(ctx, mux, marshaler, w , r, err)
```

Reply:
I was assuming they might want to handle additional statuses, but a straightforward example should be sufficient for anyone to convert to handling multiple.